pkg/settlement/swap/transaction/mock/transaction.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package mock
+
+import (
+	"context"
+	"errors"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethersphere/bee/pkg/settlement/swap/transaction"
+)
+
+type transactionServiceMock struct {
+	send           func(ctx context.Context, request *transaction.TxRequest) (txHash common.Hash, err error)
+	waitForReceipt func(ctx context.Context, txHash common.Hash) (receipt *types.Receipt, err error)
+}
+
+func (m *transactionServiceMock) Send(ctx context.Context, request *transaction.TxRequest) (txHash common.Hash, err error) {
+	if m.send != nil {
+		return m.send(ctx, request)
+	}
+	return common.Hash{}, errors.New("error")
```

Comment:
these `errors.New("error")` errors are a bit strange. throughout, why are we sabotaging. extra information?

Reply:
this is a mock so its not that bad. the errors that the service below can return are coming from the go-ethereum package if i understand correctly. not sure if the mock should use these errors or not, as the tests don't do assertions on the error type. @ralph-pichler if a mock returns an error it would be nice to be able to set the error from the test. see [topology mock](https://github.com/ethersphere/bee/blob/master/pkg/topology/mock/mock.go#L36) as an example

Reply:
Actually, as `send` function can be set as any and can return arbitrary error for checking in tests, this error is of not so significance. It can state `not implemented` or `this method is not specified`, just to be more informative.

Reply:
I went with `not implemented`. As @janos said if you want to return a specific error in the mock you can just override that function (this is also how its done in swap tests).